packages/application-shell/src/components/navbar/navbar.tsx

```diff
@@ -329,6 +348,7 @@ const RestrictedMenuItem = (props: RestrictedMenuItemProps) => {
         }}
         // Always check that some of the given permissions match.
         shouldMatchSomePermissions={true}
+        projectPermissions={props.projectPermissions}
```

Comment:
Here we explicitly pass the project permissions (specific to the menu links) to the `RestrictedByPermissions` component.